esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -1134,58 +1161,65 @@ bool EsdlBindingImpl::reloadDefinitionsFromCentralStore(IPropertyTree * esdlBndC
     return false;
 }
 
-
-bool EsdlBindingImpl::reloadBindingFromCentralStore(const char *binding, const char *process)
+bool EsdlBindingImpl::reloadBindingFromCentralStore(const char* bindingId)
 {
-    if (strcmp(m_bindingName.get(), binding)==0)
+    if(!bindingId || !*bindingId)
+        return false;
+    if(m_bindingId.length() == 0 || strcmp(m_bindingId.str(), bindingId) != 0)
+        m_bindingId.set(bindingId);
+    try
     {
-        try
+        DBGLOG("Reloading binding %s...", bindingId);
+        if ( m_pESDLService)
         {
-            DBGLOG("Reloading binding %s.%s...", m_processName.get(), m_bindingName.get());
-            if ( m_pESDLService)
-            {
-                Owned<IPropertyTree> tempEsdlBndCfg;
-                DBGLOG("Fetching ESDL binding information from dali based on ESP binding (%s.%s)", m_processName.get(), m_bindingName.get());
-                tempEsdlBndCfg.set(m_pCentralStore->fetchBinding(process, binding));
+            Owned<IPropertyTree> tempEsdlBndCfg;
+            DBGLOG("Fetching ESDL binding information from dali based on ESP binding (%s)", bindingId);
+            tempEsdlBndCfg.set(m_pCentralStore->fetchBinding(bindingId));
```

Comment:
It isn't clear from the name fetchBinding whether the pointer it returns in owned or not.  (Normally we use the convention get for linked, and query for not.).   In this case it is, so this call will leak.